klippa/src/lib.rs

```diff
@@ -600,6 +800,50 @@ pub trait Subset {
     ) -> Result<(), SubsetError>;
 }
 
+// A helper trait providing a 'subset' method for various subtables that have no associated tag
+pub(crate) trait SubsetTable {
+    /// Subset this table and write a subset version of this table into serializer
+    fn subset(&self, plan: &Plan, s: &mut Serializer) -> Result<(), SerializeErrorFlags>;
+}
+
+// A trait for subtables that require additional data in order to be subsetted
+pub(crate) trait SubsetArgs {
+    type ArgsForSubset: Default;
+}
+
+pub(crate) trait SubsetTableWithArgs: SubsetArgs {
+    fn subset_with_args(
+        &self,
+        plan: &Plan,
+        s: &mut Serializer,
+        args: &Self::ArgsForSubset,
+    ) -> Result<(), SerializeErrorFlags>;
+}
+
+pub(crate) trait SubsetTwoArgs {
+    type Arg1ForSubset: Default;
+    type Arg2ForSubset: Default;
+}
+pub(crate) trait SubsetTableWithTwoArgs: SubsetTwoArgs {
+    fn subset_with_args(
+        &self,
+        plan: &Plan,
+        s: &mut Serializer,
+        arg1: &Self::Arg1ForSubset,
+        arg2: &Self::Arg2ForSubset,
+    ) -> Result<(), SerializeErrorFlags>;
+}
+
+// A trait for subtables that require parent table's fontdata
+pub(crate) trait SubsetTableWithFontData {
```

Comment:
In general I'd see if I couldn't fold all of this into one trait, if we end up passing in the data to some tables that don't use it that seems fine?